src/sudo/pam.rs

```diff
@@ -96,6 +96,11 @@ pub(super) fn attempt_authenticate(
     non_interactive: bool,
     mut max_tries: u16,
 ) -> Result<(), Error> {
+    // Reject zero upfront so the counter never wraps on subtraction.
+    if max_tries == 0 {
+        return Err(Error::MaxAuthAttempts(0));
+    }
+
     let mut current_try = 0;
     loop {
```

Comment:
Maybe this could do `if current_try <= max_tries { return Err(Error::MaxAuthAttempts(current_try)); }` and then remove the `max_tries -= 1`?